plumbing/object/commitgraph/commitnode_graph_builder.go

```diff
@@ -0,0 +1,231 @@
+package commitgraph
+
+import (
+	"errors"
+	"math"
+
+	"github.com/go-git/go-git/v5/plumbing"
+	commitgraph "github.com/go-git/go-git/v5/plumbing/format/commitgraph/v2"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/storer"
+)
+
+type CreateCommitNodeGraphOptions struct {
+	Commits object.CommitIter // If nil then all commits in the storage are used as potential heads, otherwise only the commits in the iter are used
+	Append  bool              // Append all commits in the old index to the new index
+	Chain   bool              // Chain the new index to the old index (only valid if Append is false and the old index has generation v2)
+}
+
+// CreateCommitGraph takes a provided storage and possble old Index and creates a new commit graph.
+// To create a pack specific graph pass in a storer that is restricted to the packfile.
+func CreateCommitGraph(s storer.EncodedObjectStorer, old commitgraph.Index, opts CreateCommitNodeGraphOptions) (commitgraph.Index, error) {
+	// TODO: Implement splits
```

Comment:
Please expand on the command on why we need to do that.

Reply:
That's very odd - I'd removed this line! I think I've pushed up the wrong branch!